src/servers/src/pipeline.rs

```diff
@@ -134,8 +136,12 @@ async fn run_custom_pipeline(
             .context(PipelineSnafu)?;
 
         match r {
-            PipelineExecOutput::Transformed(row) => {
-                transformed.push(row);
+            PipelineExecOutput::Transformed((row, table_name)) => {
+                let table_name = table_name.unwrap_or(default_table.clone());
+                req_map
+                    .entry(table_name)
+                    .or_insert_with(|| Vec::with_capacity(arr_len))
```

Comment:
There may be a waste of memory here, but there is currently no way to determine the distribution of the data, so there's not much to do here at the moment, maybe in the future RowInsertRequest could accept an arbitrary type that implements IntoIterator. It would be better to use a chained table type

Reply:
@sunng87

Reply:
Here we trade space for time, I think it's fine. Doing one-time allocation is better.